backend/functions/cases/cases.function.test.ts

```diff
@@ -1,65 +1,122 @@
-    expect(httpErrorSpy).toHaveBeenCalledWith(expect.any(UnknownError));
+  afterAll(() => {
+    process.env = originalEnv;
   });
 
-  // TODO rethink how to trigger a CAMS error for this test
-  test('should call httpError if a CamsError is caught getting a case', async () => {
-    const requestOverride = {
-      ...request,
-      params: { caseId: '00000' },
-    };
+  test('Should return a case when called with a caseId', async () => {
+    const caseDetails = MockData.getCaseDetail();
+    const request = createMockAzureFunctionRequest({
+      method: 'GET',
+      params: {
+        caseId: caseDetails.caseId,
+      },
+    });
+
+    const expects = buildTestResponseSuccess<ResourceActions<CaseDetail>>({
+      data: caseDetails,
+    });
+    const { camsHttpResponse, azureHttpResponse } = expects;
 
-    const httpErrorSpy = jest.spyOn(httpResponseModule, 'httpError');
-    await httpTrigger(context, requestOverride);
+    jest.spyOn(CasesController.prototype, 'getCaseDetails').mockResolvedValue(camsHttpResponse);
 
-    expect(context.res.statusCode).toEqual(INTERNAL_SERVER_ERROR);
-    expect(context.res.body.message).toEqual('Unknown CAMS Error');
-    expect(httpErrorSpy).toHaveBeenCalledWith(expect.any(CamsError));
-    expect(httpErrorSpy).not.toHaveBeenCalledWith(expect.any(UnknownError));
+    const response = await handler(request, context);
+    expect(response).toEqual(azureHttpResponse);
   });
 
-  test('should call httpError if a CamsError is caught getting all cases', async () => {
-    const httpErrorSpy = jest.spyOn(httpResponseModule, 'httpError');
-    await httpTrigger(context, request);
+  test('should perform search', async () => {
+    const request = createMockAzureFunctionRequest({
+      method: 'POST',
+      body: {
+        caseNumber: '00-12345',
+      },
+    });
+    const expects = buildTestResponseSuccess<ResourceActions<CaseBasics>[]>({
+      data: [MockData.getCaseBasics(), MockData.getCaseBasics()],
+    });
+    const { camsHttpResponse, azureHttpResponse } = expects;
+
+    jest.spyOn(CasesController.prototype, 'searchCases').mockResolvedValue(camsHttpResponse);
+
+    const response = await handler(request, context);
+
+    expect(response).toEqual(azureHttpResponse);
+  });
+});
+
+describe('Cases function errors', () => {
+  const context = createMockAzureFunctionContext();
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+  });
+
+  type TestParams = { name: string; error: Error; requestParam: object };
+  const params: TestParams[] = [
+    {
+      name: 'getCaseDetails',
+      error: new NotFoundError('test-error'),
+      requestParam: { params: { caseId: '000-00-12345' } },
+    },
+    {
+      name: 'searchCases',
+      error: new Error('test-error'),
+      requestParam: {
+        method: 'POST',
+        body: { caseNumber: '00-12345' },
+      },
+    },
+  ];
+
+  test.each(params)(`should handle error from $name`, async (params: TestParams) => {
```

Comment:
This is interesting.  In another test we were using the %s in a template string for test.each.  Here we're using $name.  I guess using $foo is possible because we're using an array of objects here?

Reply:
Have we verified that these tests are printing the expected titles in both cases?

Reply:
![Screenshot 2024-09-04 at 3 29 16 PM](https://github.com/user-attachments/assets/d0374377-b73b-433f-8377-7b5a8905df66)